tests/CMakeLists.txt

```
@@ -38,7 +38,7 @@ set(TEST_DEPENDECIES
     foonathan::lexy
     bt_sample_nodes)
 
-if(ament_cmake_FOUND AND BUILD_TESTING)
+if(ament_cmake_FOUND)
```

I think the original check is correct

```suggestion
if(ament_cmake_FOUND AND BUILD_TESTING)
```

The problem here is that gtest was being found and linked transitively via `ament_cmake`. We'll likely need to explicitly call `find_package(ament_cmake_gtest REQUIRED)` before we call `ament_add_gtest` below and add a `<test_depend>` for the same in package.xml.

---

also, as I noted in my comments on the issue, it seems that this case of the if statement isn't even being hit on the build farm

https://github.com/BehaviorTree/BehaviorTree.CPP/issues/960#issuecomment-2803472805